decompiler/structures/pseudo/delogic_logic.py

```diff
@@ -0,0 +1,159 @@
+"""Implements translating pseudo instructions into delogic statements."""
+from __future__ import annotations
+
+from typing import Generic, Union
+
+from simplifier.world.nodes import Operation as WorldOperation
+from simplifier.world.nodes import Variable as WorldVariable
+from simplifier.world.nodes import WorldObject
+from simplifier.world.world import World
+
+from .expressions import Constant, Expression, Variable
+from .instructions import Branch
+from .logic import BaseConverter
+from .operations import Condition, Operation, OperationType
+
+
+class DelogicConverter(BaseConverter):
+    """Class in charge of converting psudo expressions into Logic statements"""
+
+    def __init__(self):
+        self._world = World()
+        self._var_count = 0
+
+    @staticmethod
+    def get_world():
+        """Retrieve a World object."""
+        return World()
+
+    def convert(self, expression: Union[Expression, Branch], w: World = None) -> WorldObject:
+        """Convert a given expression or branch into a logic statement of type WorldObject.
+        w can be obtained from LogicConverter.get_world()"""
+        if isinstance(expression, Variable):
+            return self._convert_variable(expression, self._world if w is None else w)
+        if isinstance(expression, Constant) and (isinstance(expression.value, int) or isinstance(expression.value, float)):
+            return self._convert_constant(expression, self._world if w is None else w)
+        if isinstance(expression, Branch):
+            return self._convert_branch(expression, self._world if w is None else w)
+        if isinstance(expression, Condition):
+            return self._convert_condition(expression, self._world if w is None else w)
+        if isinstance(expression, Operation):
+            return self._convert_operation(expression, self._world if w is None else w)
+        raise ValueError(f"Could not convert {expression} into a logic statement.")
+
+    def make_variable(self, expr: WorldObject) -> WorldObject:
+        """Bind a variable to an expression."""
+        var_name = f"delogic_var_{self._var_count}"
+        self._var_count += 1
+        var = self._world.variable(var_name, expr.size)
+        self._world.define(var, expr)
+        return var
+
+    def compare(self, a: WorldObject, b: WorldObject):
+        """Compare two world objects with each other."""
+        return self._world.compare(a, b)
+
+    def false(self) -> WorldObject:
+        """Return False in Delogic."""
+        return self._world.constant(0, 1)
+
+    def negate(self, expr: WorldObject) -> WorldObject:
+        """Negate a given expression."""
+        if isinstance(expr, WorldVariable):
+            return self._world.bool_negate(self._world.get_definition(expr))
+        return self._world.bool_negate(expr)
```

Comment:
I think this is not 100% what you want. It could be that expression is a variable that does not define something, i.e., a "real" variable. Then `get_definition` would return None. I am not sure whether should an expression would ever be given as input. To be sure, I would suggest the following:
```suggestion
        if isinstance(expr, WorldVariable) and formula := self._world.get_definition(expr):
            return self._world.bool_negate(formula)
        return self._world.bool_negate(expr)
```